EIPS/eip-5089.md

```diff
@@ -0,0 +1,340 @@
+author: Julian Traversa (@JTraversa), Robert Robbins (@robrobbins)
+discussions-to: https://ethereum-magicians.org/t/eip-xxxx-principal-token-standard/1
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-05-01
+requires: 20, 2612
+---
+
+## Abstract
+
+The following standard allows for the implementation of a standard API for principal tokens --
+representing ownership of an underlying [ERC-20](./eip-20.md) token at a future timestamp.
+This standard is an extension on the ERC-20 token that provides basic functionality for depositing
+and withdrawing tokens and reading balances and the ERC-2612 standard that provides
+EIP-712 signature based approvals.
+
+## Motivation
+
+Principal tokens have an increasing lack of standardization which has led to a difficult to navigate development space and diverse implementation
+schemes.
+
+The primary examples include yield tokenization platforms which strip future yield leaving a principal
+token behind, as well as fixed-rate money-markets which utilize principal tokens as a medium
+to lend/borrow.
+
+This inconsistency in implementation makes integration difficult at the application layer as well as
+wallet layer which are key catalysts for the space's growth. 
+Developers are currently expected to implement individual adapters for each principal token, as well as adapters for
+their pool contracts, and many times adapters for their custodial contracts as well, wasting significant developer resources. 
+
+## Specification
+
+All ERC-5070 Principal Tokens MUST implement ERC-20 to represent ownership of future underlying redemption (PTs).
+If a PT is to be non-transferrable, it MAY revert on calls to `transfer` or `transferFrom`.
+The ERC-20 operations `balanceOf`, `transfer`, `totalSupply`, etc. operate on the Principal Token balance
+which represents a claim to ownership of 1:1 amounts of underlying upon maturity.
+
+All ERC-5070 Principal Tokens MUST implement ERC-20's optional metadata extensions.
+The `name` and `symbol` functions SHOULD reflect the underlying token's `name` and `symbol` in some way, as well as the origination protocol, 
+and in the case of yield tokenization protocols, the origination money-market.
+
+ERC-5070 tokenized Vaults MAY implement [EIP-2612](./eip-2612.md) to improve the UX of approving PTs on various integrations.
+
+### Definitions:
+
+- underlying: The token that Principal Tokens are redeemable for at maturity.
+  Has units defined by the corresponding ERC-20 contract.
+- maturity: The timestamp (unix) at which a Principal Token matures. Principal Tokens become redeemable for underlying at or after this timestamp.
+- exchangeRate: A variable used to modulate the redeemable ratio of underlying <-> Principal Token.
+- fee: An amount of underlying or Principal Token charged to the user by the Principal Token. Fees can exists on redemption or post-maturity yield. _Alberto: I'm not sure we need this in the ERC._
```

Comment:
What were the arguments for it being in 4626?

I didn't think it deserved to be there either but it seemed like folks wanted to futureproof the standard. Contextually it makes even less sense here but just curious.

Reply:
In 4626 it was assumed that some implementations would definitely include fees on deposits or withdrawals. I agree that maybe fees don't make much sense for principal tokens.